File: 2259-remove-digit-from-number-to-maximize-result/2259-remove-digit-from-number-to-maximize-result.cpp
class Solution {
public:
    string removeDigit(string number, char digit) {
        string mx="";
        
        for(int i=0;i<number.size();i++){
            if(number[i] == digit){
                string str = number.substr(0,i)+number.substr(i+1);
                mx=max(mx,str);
            }
        }
        return mx;
    }
};

File: 2259-remove-digit-from-number-to-maximize-result/2259-remove-digit-from-number-to-maximize-result.js
/**
 * @param {string} number
 * @param {character} digit
 * @return {string}
 */
var removeDigit = function(number, digit) {
    let mx = 0n; // Initialize mx as a BigInt since we'll be comparing BigInt values

    for (let i = 0; i < number.length; i++) {
        if (number[i] == digit) {
            let temp = number.slice(0, i) + number.slice(i + 1);
           let tempBigInt = BigInt(temp);
            mx = mx > tempBigInt ? mx : tempBigInt;
        }
    }

    return mx.toString();
};